tests/cases/compiler/baseClassExpressionOutOfOrder.ts

```diff
@@ -0,0 +1,15 @@
+var c = class A extends B { // error
+}

```

Comment:
i would also add a test with an expression in an extend clause, `class A extends foo(0)<string> {}`
and one for module aliases 
```ts
import d from "mod"; 
class B extends d { // no error here
}
```